diff --git a/tests/command_line/test_convert_to_cbf.py b/tests/command_line/test_convert_to_cbf.py
index 1dc5d01c9..82115220e 100644
--- a/tests/command_line/test_convert_to_cbf.py
+++ b/tests/command_line/test_convert_to_cbf.py
@@ -4,6 +4,7 @@
 import pytest
 
 
+@pytest.mark.xfail
 @pytest.mark.parametrize("filename", ["image_15799_master.h5", "image_15799.nxs"])
 def test_convert_to_cbf(dials_data, filename, tmp_path):
     result = procrunner.run(
