diff --git a/tests/command_line/test_rs_mapper.py b/tests/command_line/test_rs_mapper.py
index c08e48797..909d15a4c 100644
--- a/tests/command_line/test_rs_mapper.py
+++ b/tests/command_line/test_rs_mapper.py
@@ -11,6 +11,7 @@ def test_rs_mapper(dials_data, tmp_path):
     result = procrunner.run(
         [
             "dials.rs_mapper",
+            "nproc=1",
             dials_data("centroid_test_data", pathlib=True)
             / "imported_experiments.json",
             'map_file="junk.ccp4"',
@@ -42,7 +43,7 @@ def test_masked(dials_data, tmp_path):
         working_directory=tmp_path,
     )
     result = procrunner.run(
-        ["dials.rs_mapper", "imported.expt", "map_file=masked.ccp4"],
+        ["dials.rs_mapper", "nproc=1", "imported.expt", "map_file=masked.ccp4"],
         working_directory=tmp_path,
     )
     assert not result.returncode and not result.stderr
@@ -52,6 +53,7 @@ def test_masked(dials_data, tmp_path):
     result = procrunner.run(
         [
             "dials.rs_mapper",
+            "nproc=1",
             "imported.expt",
             "map_file=unmasked.ccp4",
             "ignore_mask=True",
