diff --git a/src/dials/algorithms/integration/integrator.py b/src/dials/algorithms/integration/integrator.py
index fc7003c6a..264a091e9 100644
--- a/src/dials/algorithms/integration/integrator.py
+++ b/src/dials/algorithms/integration/integrator.py
@@ -616,6 +616,8 @@ class ProfileModellerExecutor(Executor):
     The class to do profile modelling calculations
     """
 
+    __getstate_manages_dict__ = 1
+
     def __init__(self, experiments, profile_fitter):
         """
         Initialise the executor
@@ -719,6 +721,8 @@ class ProfileValidatorExecutor(Executor):
     The class to do profile validation calculations
     """
 
+    __getstate_manages_dict__ = 1
+
     def __init__(self, experiments, profile_fitter):
         """
         Initialise the executor
@@ -822,6 +826,8 @@ class IntegratorExecutor(Executor):
     The class to process the integration data
     """
 
+    __getstate_manages_dict__ = 1
+
     def __init__(
         self, experiments, profile_fitter=None, valid_foreground_threshold=0.75
     ):
