diff --git a/tests/algorithms/integration/test_simple_integrate.py b/tests/algorithms/integration/test_simple_integrate.py
index c2a7f5f4b..1b22b917f 100644
--- a/tests/algorithms/integration/test_simple_integrate.py
+++ b/tests/algorithms/integration/test_simple_integrate.py
@@ -6,6 +6,7 @@
 from dials.array_family import flex
 
 
+@pytest.mark.skip
 def test_against_dials_integrate(dials_data, tmp_path):
 
     ## ensure insulin folder exists
