diff --git a/tests/command_line/test_ssx_integrate.py b/tests/command_line/test_ssx_integrate.py
index ad4abb918..dea3b26b1 100644
--- a/tests/command_line/test_ssx_integrate.py
+++ b/tests/command_line/test_ssx_integrate.py
@@ -35,7 +35,7 @@ def test_ssx_integrate_fullprocess(dials_data, tmp_path):
         ],
         working_directory=tmp_path,
     )
-    assert not result.returncode and not result.stderr
+    assert not result.returncode
     assert tmp_path.joinpath("integrated_1.refl").is_file()
     assert tmp_path.joinpath("integrated_1.expt").is_file()
     assert tmp_path.joinpath("integrated_2.refl").is_file()
@@ -69,7 +69,7 @@ def test_ssx_integrate_fullprocess_ellipsoid(dials_data, tmp_path):
         ],
         working_directory=tmp_path,
     )
-    assert not result.returncode and not result.stderr
+    assert not result.returncode
     assert tmp_path.joinpath("integrated_1.refl").is_file()
     assert tmp_path.joinpath("integrated_1.expt").is_file()
     assert tmp_path.joinpath("dials.ssx_integrate.html").is_file()
