diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2856843c84..b899b5f1d2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,8 +17,14 @@ set(CMAKE_CXX_STANDARD 14)
 find_package(Python REQUIRED COMPONENTS Interpreter Development)
 find_package(CCTBX COMPONENTS cctbx scitbx ccp4io annlib REQUIRED)
 find_package(DXTBX REQUIRED)
-find_package(msgpack REQUIRED)
 find_package(OpenMP)
+# Handle msgpack.. between versions 3 and 6, split into separate packages.
+# While we still support both, look for both names
+find_package(msgpack-cxx REQUIRED NAMES msgpack-cxx msgpack)
+if (TARGET msgpackc AND NOT TARGET msgpack-cxx)
+    # If the old version of the library, let us refer to the new name
+    add_library(msgpack-cxx ALIAS msgpackc)
+endif()
 
 # Find the boost::python library for this version of python
 set(Boost_USE_STATIC_LIBS OFF) # This is the default everywhere except Windows
diff --git a/src/dials/array_family/CMakeLists.txt b/src/dials/array_family/CMakeLists.txt
index cd8279993e..2d28ac9b65 100644
--- a/src/dials/array_family/CMakeLists.txt
+++ b/src/dials/array_family/CMakeLists.txt
@@ -18,5 +18,5 @@ target_link_libraries(
     CCTBX::cctbx
     Boost::python
     CCTBX::scitbx::boost_python
-    msgpackc
+    msgpack-cxx
 )
