diff --git a/tests/command_line/test_model_background.py b/tests/command_line/test_model_background.py
index ad887c839..5d1a53995 100644
--- a/tests/command_line/test_model_background.py
+++ b/tests/command_line/test_model_background.py
@@ -8,6 +8,7 @@
 from scitbx.array_family import flex
 
 
+@pytest.skip(reason="Apparently causes SEGV on some platforms")
 def test_model_background(dials_data, tmp_path):
     centroid = dials_data("centroid_test_data", pathlib=True)
     expts = centroid / "experiments.json"
